Arm/Cosim.lean

```diff
@@ -205,19 +251,33 @@ def nzcv_mismatch (x1 x2 : BitVec 4) : IO String := do
       acc := acc ++ s!"Flag{flag} machine {f1} model {f2}"
   pure acc
 
-def regStates_match (uniqueBaseName : String) (input o1 o2 : regState) : 
+def regStates_match (uniqueBaseName : String) (input o1 o2 : regState) :
   IO Bool := do
   if o1 == o2 then
-     pure true
+      pure true
   else
-     let gpr_mismatch ← gpr_mismatch o1.gpr o2.gpr
+    let darwin_check ←
+    IO.Process.output
+      { cmd  := "Arm/Insts/Cosim/platform_check.sh",
+        args := #["-d"] }
+    let isDarwin := (darwin_check.exitCode == 1)
+     let gpr_mismatch ← gpr_mismatch isDarwin o1.gpr o2.gpr
      let nzcv_mismatch ← nzcv_mismatch o1.nzcv o2.nzcv
-     let sfp_mismatch  ← sfp_mismatch o1.sfp o2.sfp
-     IO.println s!"ID: {uniqueBaseName}"
-     IO.println s!"Instruction: {decode_raw_inst input.inst}"
-     IO.println s!"input: {toString input}"
-     IO.println s!"Mismatch found: {gpr_mismatch} {nzcv_mismatch} {sfp_mismatch}"
-     pure false
+     let sfp_mismatch  ← sfp_mismatch o1.sfp o2.sfp     
```

Comment:
The `o1 == o2` case is covered by the following if condition. Maybe we could remove the `o1 == o2` check?

Reply:
I expect this check to be inexpensive compared the following one --- it's there to hopefully simplify the comparison on non-Apple Arm machines.